CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+# Contributing
+
+**Table of Contents**
+1. [Welcome](#welcome) (2nd draft)
+2. [Philosophy](#philosophy) (TBD)
+3. [Reporting an Issue](#reporting-an-issue) (TBD)
+4. [Fixing an Issue (PRs)](#fixing-an-issue-prs) (2nd draft)
+5. [Setup and Run](#setup-and-run) (TBD)
+
+
+## Welcome
+Hi, and welcome to Highlight.js. We're a code-highlighting engine etc., but you probably know that. You want to contribute, and this document will be your guide.
+
+There are many ways to contribute to Highlight.js. Reporting issues is super useful. Feel free to work on an open issue. You can also help by writing [a plugin](https://highlightjs.readthedocs.io/en/latest/plugin-api.html), adding [a 3rd party grammar](https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md), or creating [a theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html).
+
+In terms of getting help, GitHub issues are the primary method of communication. We're exploring the idea of creating a Gitter chat room.
+
+
+## Philosophy
```

Comment:
A bit long, imho. People need to actually _read_ it 😁.

Reply:
Really?  I didn't find it that along... I can take a second look... did you imagine just a bullet point list or?   What length do you imagine is "perfect" for such a thing?

Reply:
multiple sections is fine. I'm mainly worried I usually don't read documents on the web word-for-word, so it should be a easy to scan.